### PR TITLE
Centralize warning suppression

### PR DESCRIPTION
**Pull request by glenn-jocher**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Centralizes warning suppression across Ultralytics to reduce noisy logs and simplify exporter, plotting, MobileCLIP, and TensorBoard code 🧹🔕.

### 📊 Key Changes
- Moved scattered `warnings.filterwarnings(...)` calls into a single, global suppression block in `ultralytics/utils/__init__.py` 🔧
- Removed local warning suppression from:
  - Export pipeline (`ultralytics/engine/exporter.py`) 🧠➡️📦
  - MobileCLIP text model import (`ultralytics/nn/text_model.py`) 📝
  - Label plotting (`ultralytics/utils/plotting.py`) 📈
  - TensorBoard callback module imports (`ultralytics/utils/callbacks/tensorboard.py`) 📊
- Improved TensorBoard graph logging flow:
  - Uses a clearer two-stage attempt (simple trace first, then RT-DETR-style export fallback)
  - Logs a more informative failure message showing both errors if both attempts fail 🧾✅⚠️

### 🎯 Purpose & Impact
- Cleaner, more maintainable code by avoiding repeated per-file warning filters 🧼
- More consistent user experience: common third-party warnings (PyTorch tracer, ONNX shape warnings, timm FutureWarnings, CoreML deprecations, matplotlib layout messages) are suppressed uniformly 🔇
- Better TensorBoard debugging: when graph logging fails, users get clearer context about what went wrong 📌
- Potential tradeoff: centralized suppression applies globally, which may hide warnings some advanced users relied on for debugging 🕵️‍♂️

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist (this is automated—an engineer will assist shortly) 🤖✨:

-✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
-✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
-✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
-✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
-✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
-✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
-✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/utils/__init__.py [hunk 1]
```diff
@@ -132,6 +133,13 @@
 os.environ["TORCH_CPP_LOG_LEVEL"] = "ERROR"  # suppress "NNPACK.cpp could not initialize NNPACK" warnings
 os.environ["KINETO_LOG_LEVEL"] = "5"  # suppress verbose PyTorch profiler output when computing FLOPs
 
+# Centralized warning suppression
+warnings.filterwarnings("ignore", message="torch.distributed.reduce_op is deprecated")  # PyTorch deprecation
+warnings.filterwarnings("ignore", message="The figure layout has changed to tight")  # matplotlib>=3.7.2
+warnings.filterwarnings("ignore", message=".*is deprecated and slated for removal.*")  # mobileclip and others
+warnings.filterwarnings("ignore", category=torch.jit.TracerWarning)  # ONNX/TorchScript export tracer warnings
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/__init__.py:140]: ❗ **CRITICAL**: `torch` is referenced here (`torch.jit.TracerWarning`) but this diff does not show `torch` being imported before this block executes. If `torch` is imported later in the module, this will raise a `NameError` during `ultralytics.utils` import and break all downstream imports. Please ensure `torch` is imported before these filters (or gate these filters behind a successful torch import).

File: ultralytics/utils/__init__.py [hunk 2]
```diff
@@ -132,6 +133,13 @@
 os.environ["TORCH_CPP_LOG_LEVEL"] = "ERROR"  # suppress "NNPACK.cpp could not initialize NNPACK" warnings
 os.environ["KINETO_LOG_LEVEL"] = "5"  # suppress verbose PyTorch profiler output when computing FLOPs
 
+# Centralized warning suppression
+warnings.filterwarnings("ignore", message="torch.distributed.reduce_op is deprecated")  # PyTorch deprecation
+warnings.filterwarnings("ignore", message="The figure layout has changed to tight")  # matplotlib>=3.7.2
+warnings.filterwarnings("ignore", message=".*is deprecated and slated for removal.*")  # mobileclip and others
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/__init__.py]: 💡 **MEDIUM**: This message-based suppression is very broad (no `category`/`module`), and could end up hiding unrelated deprecations across the entire package (including user code) as long as the warning message matches. Consider scoping it by `category` (e.g., `FutureWarning`/`DeprecationWarning`) and/or `module` to avoid masking meaningful warnings.

File: ultralytics/utils/__init__.py [hunk 3]
```diff
@@ -132,6 +133,13 @@
...
+warnings.filterwarnings("ignore", category=torch.jit.TracerWarning)  # ONNX/TorchScript export tracer warnings
+warnings.filterwarnings("ignore", category=UserWarning, module="torch.jit")  # JIT trace warnings
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/__init__.py]: 💡 **MEDIUM**: The previous local suppressions were not limited to `module="torch.jit"` (e.g., ONNX export warnings can be emitted outside `torch.jit`). Centralizing is good, but this may be too narrow and could reintroduce noisy warnings in exporter/graph logging. Consider expanding/remapping this filter to cover the actual warning origin you’re targeting.

File: ultralytics/utils/callbacks/tensorboard.py [hunk 4]
```diff
@@ -61,32 +60,27 @@ def _log_tensorboard_graph(trainer) -> None:
...
-        warnings.simplefilter("ignore", category=torch.jit.TracerWarning)  # suppress jit trace warning
-
-        # Try simple method first (YOLO)
+    # Try simple method first (YOLO)
+    try:
+        trainer.model.eval()  # place in .eval() mode to avoid BatchNorm statistics changes
+        WRITER.add_graph(torch.jit.trace(torch_utils.unwrap_model(trainer.model), im, strict=False), [])
+        LOGGER.info(f"{PREFIX}model graph visualization added ✅")
+        return
+    except Exception:
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/callbacks/tensorboard.py]: 📝 **LOW**: Catching `Exception` here without logging the first failure makes debugging graph logging issues harder (the fallback may also fail for a different reason). Consider capturing the first exception and including it in the warning if the fallback fails, or at least logging it at debug level.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Good consolidation effort and the TensorBoard tracing flow remains intact. Main blocker: the centralized suppression block in `ultralytics/utils/__init__.py` appears to reference `torch` before it’s guaranteed to be imported, which can break imports at runtime. Also, one suppression is overly broad and another may be too narrowly scoped to `torch.jit`, potentially reintroducing warnings elsewhere.

💬 Posted 4 inline comments

**Comment by UltralyticsAssistant** [comment]
Thrilled to see this merged—thank you, @glenn-jocher! Centralizing warning suppression into `ultralytics/utils/__init__.py` is a clean, high-leverage refactor that makes the exporter, plotting, MobileCLIP, and TensorBoard paths noticeably calmer and easier to maintain, while the improved TensorBoard two-stage graph logging (and clearer dual-error reporting) is a real quality-of-life win for debugging.

As Leonardo da Vinci said, **“Simplicity is the ultimate sophistication.”** This PR embodies that: fewer scattered filters, more consistent behavior, and a smoother developer + user experience across Ultralytics.

Appreciate the steady attention to polish that keeps YOLO11 workflows fast, readable, and reliable. 🙌
